src/nix-eval-jobs.cc

```diff
@@ -231,17 +241,24 @@ static void worker(
 
             else if (v->type() == nAttrs)
               {
-                auto attrs = nlohmann::json::array();
-                StringSet ss;
-                for (auto & i : v->attrs->lexicographicOrder()) {
-                    std::string name(i->name);
-                    if (name.find('.') != std::string::npos || name.find(' ') != std::string::npos) {
-                        printError("skipping job with illegal name '%s'", name);
-                        continue;
+                // Hacky way to get depth.... fails with embedded "."
+                size_t n = std::count(attrPath.begin(), attrPath.end(), '.')+(attrPath == "" ? 0: 1);
+                if (n < myArgs.depth){
+                    auto attrs = nlohmann::json::array();
+                    Bindings::iterator j = v->attrs->find(state.sRecurseForDerivations);
```

Comment:
Can you make a test case for this or just extend this file? https://github.com/nix-community/nix-eval-jobs/blob/4bab9e449da80b58507d8d3ab6c051701e8d65e4/tests/assets/ci.nix#L6

Reply:
Otherwise looks good to me.